gno.land/pkg/sdk/vm/gas_test.go

```diff
@@ -0,0 +1,176 @@
+package vm
+
+import (
+	"testing"
+
+	bft "github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto"
+	"github.com/gnolang/gno/tm2/pkg/sdk"
+	"github.com/gnolang/gno/tm2/pkg/sdk/auth"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/gnolang/gno/tm2/pkg/store"
+	"github.com/jaekwon/testify/assert"
+)
+
+// Gas is consumed in both CheckTx and DeliverTx.
+// VM gas is consumed only in DeliverTx.
```

Comment:
this comment seems strange. 
gas costs are deducted in both checktx and delivertx.
gas used (VM, storage, bytes) is only checked in delivertx.
right?

Reply:
Yes, the entire gas amount is deducted in both checkTx and deliverTx. However, gas used for VM messages (CPU and store access by bytes) is consumed (deducted) only in deliverTx. Gas used for balance checking, message size, and signature verification is consumed (deducted) in checkTx.